aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -1249,6 +1316,51 @@ private boolean s3KeyIdAccessKeyBothConfigured() {
     return (s3AccessKeyId == null) == (s3SecretAccessKey == null);
   }
 
+  private AwsCredentialsProvider credentialsProvider(String credentialsProviderClazz) {
+    try {
+      Class<?> providerClass =
+          DynClasses.builder()
+              .loader(this.getClass().getClassLoader())
+              .impl(credentialsProviderClazz)
+              .buildChecked();
+
+      ValidationException.check(
```

Comment:
This is the wrong exception type, we throw these only for table validation errors. 
```java
/** 
 * Exception raised when validation checks fail.
 * <p>
 * For example, this is thrown when attempting to create a table with a {@link PartitionSpec} that
 * is not compatible with the table {@link Schema}
 */
 ```
 
 I think this is probably better as a `Preconditions.checkArg`

Reply:
Replaced ValidationException.check with Preconditions.  I also noticed that AwsProperties class still has ValidationException.check API call 

https://github.com/apache/iceberg/blob/master/aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java#L777